File: Payload_Type/thanatos/mythic/command/pwd/pwd.go
package commands

import (
	agentstructs "github.com/MythicMeta/MythicContainer/agent_structs"
)

var pwdCommandDefinition = agentstructs.Command{
	Name:        "pwd",
	Description: "Print the current working directory.",
	HelpString:  "pwd",
	Version:     1,

	MitreAttackMappings: []string{
		"T1005",
		"T1039",
		"T1025",
	},

	Author: "@M_alphaaa",

	SupportedUIFeatures: []string{"callback_table:pwd"},

	CommandAttributes: agentstructs.CommandAttribute{
		SupportedOS: []string{
			agentstructs.SUPPORTED_OS_LINUX, agentstructs.SUPPORTED_OS_WINDOWS,
		},
	},

	TaskFunctionParseArgString: func(args *agentstructs.PTTaskMessageArgsData, input string) error {
		return nil
	},

	TaskFunctionParseArgDictionary: func(args *agentstructs.PTTaskMessageArgsData, input map[string]interface{}) error {
		return nil
	},

	TaskFunctionCreateTasking: func(task *agentstructs.PTTaskMessageAllData) agentstructs.PTTaskCreateTaskingMessageResponse {
		response := agentstructs.PTTaskCreateTaskingMessageResponse{
			Success: true,
			TaskID:  task.Task.ID,
		}

		return response
	},
}

func init() {
	agentstructs.AllPayloadData.Get("thanatos").AddCommand(pwdCommandDefinition)
}

File: Payload_Type/thanatos/mythic/thanatos/commands/pwd.py
from mythic_container.MythicCommandBase import (
    TaskArguments,
    CommandBase,
    CommandAttributes,
    SupportedOS,
    PTTaskMessageAllData,
    PTTaskCreateTaskingMessageResponse,
)


class PwdArguments(TaskArguments):
    def __init__(self, command_line, **kwargs):
        super().__init__(command_line, **kwargs)
        self.args = []

    async def parse_arguments(self):
        pass


class PwdCommand(CommandBase):
    cmd = "pwd"
    needs_admin = False
    help_cmd = "pwd"
    description = "Print the working directory."
    version = 2
    author = "@M_alphaaa"
    argument_class = PwdArguments
    supported_ui_features = ["callback_table:pwd"]
    attackmapping = ["T1083"]
    attributes = CommandAttributes(
        supported_os=[SupportedOS.Linux, SupportedOS.Windows],
    )

    async def create_go_tasking(
        self, task_data: PTTaskMessageAllData
    ) -> PTTaskCreateTaskingMessageResponse:
        return PTTaskCreateTaskingMessageResponse(TaskID=task_data.Task.ID, Success=True)
